# inputs
duration time_to_wait
---
# outputs
duration time_elapsed
uint32 updates_sent
---
# feeback
duration time_elapsed
duration time_remaining